### DISCUSSION TITLE
Error: No stream source provided

### DISCUSSION THREAD

**Discussion by SathvikKS** [Androz2091/discord-player#824] [General]
**Describe the bug**
I am trying to manually play the song by creating a track object and sending it to `queue.addTrack()`. This does trigger the `trackAdd` event and I get the `Added to queue` message, but doesnt play the song. This is the code I've used
```
const {
    Track
} = require('discord-player');
const queue = client.player.createQueue(interaction.guild, {
    leaveOnEnd: false,
    leaveOnStop: false,
    leaveOnEmpty: true,
    leaveOnEmptyCooldown: 10000,
    metadata: {
        channel: interaction.channel
    }
});
try {
    if (!queue.connection) await queue.connect(interaction.member.voice.channel);
} catch (e) {
    console.log(e)
}
}
const newTrack = new Track(client.player, {
    "id": '892389134012907522',
    "title": "I'm A Wanted Man (Official Audio) | Royal Deluxe [UFC 229 Teaser - McGregor vs. Khabib Placement]",
    "author": 'Position Music',
    "url": 'https://www.youtube.com/watch?v=YSAMOBVncSg',
    "thumbnail": 'https://i3.ytimg.com/vi/YSAMOBVncSg/maxresdefault.jpg',
    "duration": '3:50',
    "views": 9685475,
    "playlist": undefined
})
queue.addTrack(newTrack)
if (!queue.playing) await queue.play();
```
Manually create a track and play it

**Screenshots**
![image](https://user-images.githubusercontent.com/31741267/135266416-330becaf-52bc-480a-a37f-9ce5fb4e7363.png)
![image](https://user-images.githubusercontent.com/31741267/135266476-fc67c9d3-53ed-46ff-98dc-e8a87a181f9e.png)

 - Node Version: 16.10.0
 - Discord Player Version: 5.1.0
 - Discord.js Version: 13.1.0

**Comment by twlite** [comment]
Why are you manually creating track like that? Also it's not a bug, you need to specify from what source the track was created, like:

```js
const newTrack = new Track(client.player, {
    "id": '892389134012907522',
    "title": "I'm A Wanted Man (Official Audio) | Royal Deluxe [UFC 229 Teaser - McGregor vs. Khabib Placement]",
    "author": 'Position Music',
    "url": 'https://www.youtube.com/watch?v=YSAMOBVncSg',
    "thumbnail": 'https://i3.ytimg.com/vi/YSAMOBVncSg/maxresdefault.jpg',
    "duration": '3:50',
    "views": 9685475,
    "playlist": undefined,
    "source": "youtube"
})
queue.addTrack(newTrack);
```

**Comment by twlite** [comment]
and keep in mind, this is not a recommended way of doing it. You should rather use the `search` method of `Player`

**Comment by SathvikKS** [comment]
Thanks for the help. I wanted to save the tracks in this format in database and then use it to play cause I was trying to add a save playlists command.
